src/Cake.GitHub/Releases/GitHubReleaseCreator.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Cake.Core.Diagnostics;
+using Cake.Core.IO;
+using Octokit;
+
+namespace Cake.GitHub
+{
+    internal sealed class GitHubReleaseCreator
+    {
+        private readonly ICakeLog _cakeLog;
+        private readonly IFileSystem _fileSystem;
+        private readonly IGitHubClient _githubClient;
+        
+        
+        public GitHubReleaseCreator(ICakeLog cakeLog, IFileSystem fileSystem, IGitHubClient githubClient)
+        {
+            _cakeLog = cakeLog ?? throw new ArgumentNullException(nameof(cakeLog));
```

Comment:
the throws can be removed as well, nothing will ever be null here.

Reply:
Fixed